spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteUpdate.scala

```diff
@@ -74,12 +109,14 @@ case class RewriteUpdate(spark: SparkSession) extends Rule[LogicalPlan] with Rew
     val assignedExprs = assignments.map(_.value)
     val updatedExprs = assignedExprs.zip(relation.output).map { case (assignedExpr, attr) =>
       // use semanticEquals to avoid unnecessary if expressions as we may run after operator optimization
-      val updatedExpr = if (attr.semanticEquals(assignedExpr)) {
+      if (attr.semanticEquals(assignedExpr)) {
         attr
+      } else if (cond == Literal.TrueLiteral) {
```

Comment:
Is `==` always going to work? What if we get `Literal(true)`?

Reply:
It is Scala, so that would be translated as `equals`. We should be safe.

Reply:
Oh, right. Thanks!